linera-execution/solidity/Linera.sol

```diff
@@ -7,123 +7,177 @@ import "./LineraTypes.sol";
-        bytes memory input1 = new bytes(2);
-        input1[0] = bytes1(uint8(0));
-        input1[1] = bytes1(val);
-        (bool success, bytes memory output1) = precompile.call(input1);
+        LineraTypes.BaseRuntimePrecompile memory base = LineraTypes.BaseRuntimePrecompile_case_chain_id();
+        LineraTypes.RuntimePrecompile memory input1 = LineraTypes.RuntimePrecompile_case_base(base);
+        bytes memory input2 = LineraTypes.bcs_serialize_RuntimePrecompile(input1);
+        (bool success, bytes memory output) = precompile.call(input2);
         require(success);
-        return LineraTypes.bcs_deserialize_ChainId(output1);
-    }
-
-    function chain_id() internal returns (LineraTypes.ChainId memory) {
-        return inner_chain_id(uint8(0));
+        return LineraTypes.bcs_deserialize_ChainId(output);
     }
 
     function application_creator_chain_id() internal returns (LineraTypes.ChainId memory) {
-        return inner_chain_id(uint8(1));
+        address precompile = address(0x0b);
+        LineraTypes.BaseRuntimePrecompile memory base = LineraTypes.BaseRuntimePrecompile_case_application_creator_chain_id();
+        LineraTypes.RuntimePrecompile memory input1 = LineraTypes.RuntimePrecompile_case_base(base);
+        bytes memory input2 = LineraTypes.bcs_serialize_RuntimePrecompile(input1);
+        (bool success, bytes memory output) = precompile.call(input2);
+        require(success);
+        return LineraTypes.bcs_deserialize_ChainId(output);
     }
 
     function chain_ownership() internal returns (LineraTypes.ChainOwnership memory) {
         address precompile = address(0x0b);
-        bytes memory input1 = new bytes(2);
-        input1[0] = bytes1(uint8(0));
-        input1[1] = bytes1(uint8(2));
-        (bool success, bytes memory output1) = precompile.call(input1);
+        LineraTypes.BaseRuntimePrecompile memory base = LineraTypes.BaseRuntimePrecompile_case_chain_ownership();
+        LineraTypes.RuntimePrecompile memory input1 = LineraTypes.RuntimePrecompile_case_base(base);
+        bytes memory input2 = LineraTypes.bcs_serialize_RuntimePrecompile(input1);
+        (bool success, bytes memory output) = precompile.call(input2);
         require(success);
-        return LineraTypes.bcs_deserialize_ChainOwnership(output1);
+        return LineraTypes.bcs_deserialize_ChainOwnership(output);
     }
 
     function read_data_blob(bytes32 hash) internal returns (bytes memory) {
         address precompile = address(0x0b);
-        bytes memory input1 = new bytes(2);
-        input1[0] = bytes1(uint8(0));
-        input1[1] = bytes1(uint8(3));
-        bytes memory input2 = abi.encodePacked(input1, hash);
-        (bool success, bytes memory output1) = precompile.call(input2);
+        LineraTypes.CryptoHash memory hash2 = LineraTypes.CryptoHash(hash);
+	LineraTypes.BaseRuntimePrecompile_ReadDataBlob memory read_data_blob_ = LineraTypes.BaseRuntimePrecompile_ReadDataBlob(hash2);
```

Comment:
Formatting: Some of these are tabs, some are spaces.

Reply:
Yes, there was one. Just one.
Maybe we should detect tabs in CI.